src/build.ts

```diff
@@ -53,274 +32,57 @@ process.on('unhandledRejection', (reason: any) => {
-    const rust = process.argv.includes('--rust');
-    const wasi2ic = process.argv.includes('--wasi2ic');
-
-    if (!node && !dfx && !rust && !wasi2ic) {
-        await runDevSetupCommand({
-            dfx: true,
-            node: true,
-            rust: true,
-            wasi2ic: true
-        });
+    if (await isExperimental()) {
+        const { build } = await import('#experimental/build/index');
+        await build();
     } else {
-        await runDevSetupCommand({ dfx, node, rust, wasi2ic });
+        const { build } = await import('#build/index');
+        await build();
     }
 }
 
-async function handleDevTemplateCommand(ioType: IOType): Promise<void> {
-    const all = process.argv.includes('--all');
-
-    if (all === true) {
-        await runDevTemplateCommand(ioType);
-        await runExperimentalDevTemplateCommand(ioType);
-    } else {
-        const experimental =
-            process.argv.includes('--experimental') ||
-            process.env.AZLE_EXPERIMENTAL === 'true';
-
-        if (experimental === false) {
-            await runDevTemplateCommand(ioType);
-        } else {
-            await runExperimentalDevTemplateCommand(ioType);
-        }
+/**
+ * Determines if the current build should use experimental features.
+ * Checks canister config, command line arguments, and environment variables.
+ * @param canisterName The name of the canister being built.
+ * @returns A promise that resolves to true if the build is experimental, false otherwise.
+ */
+async function isExperimental(): Promise<boolean> {
+    if (process.argv.includes('--experimental') === true) {
+        return true;
     }
-}
-
-async function handleUploadAssetsCommand(): Promise<void> {
-    const canisterName = process.argv[3];
-    const canisterConfig = await getCanisterConfig(canisterName);
 
-    const experimental =
-        canisterConfig?.custom?.experimental === true ||
-        process.env.AZLE_EXPERIMENTAL === 'true';
-
-    if (experimental === false) {
-        if (canisterConfig.custom?.assets !== undefined) {
-            throw new Error(
-                experimentalMessageDfxJson('the upload-assets command')
-            );
-        }
-    } else {
-        await runExperimentalUploadAssetsCommand();
+    if (process.env.AZLE_EXPERIMENTAL === 'true') {
+        return true;
     }
-}
-
-async function handleBuildCommand(ioType: IOType): Promise<void> {
-    const canisterName = process.argv[3];
-    const canisterConfig = await getCanisterConfig(canisterName);
 
-    const experimental =
-        canisterConfig?.custom?.experimental === true ||
-        process.env.AZLE_EXPERIMENTAL === 'true';
+    const command = process.argv[2] as
+        | Command
+        | ExperimentalCommand
+        | undefined;
 
-    if (experimental === false) {
-        checkForExperimentalDfxJsonFields(canisterConfig);
+    if (
+        command === 'build' ||
+        command === 'post-install' ||
+        command === 'upload-assets'
```

Comment:
We need to somehow tell the user that upload-assets can only run in experimental mode. Right now it will just fail in an error saying the upload-assets command does not exist unless there is an experimental flag or environment variable or the flag in dfx.json...

Reply:
done